cmd/package_managers_test.go

```diff
@@ -706,3 +709,2268 @@ func Test_fetchGitRepositoryFromRubyGems(t *testing.T) {
+		                        "@type": "Package",
+		                        "catalogEntry": {
+		                            "@id": "https://api.nuget.org/v3/catalog0/data/2022.12.08.16.43.03/Foo.NET.3.5.8.json",
+		                            "@type": "PackageDetails",
+		                            "listed": true,
+		                            "version": "3.5.8"
+		                        }
+		                    },
+		                    {
+		                        "@id": "https://api.nuget.org/v3/registration5-semver1/Foo.NET/4.0.1.json",
+		                        "@type": "Package",
+		                        "catalogEntry": {
+		                            "@id": "https://api.nuget.org/v3/catalog0/data/2022.12.08.16.43.03/Foo.NET.4.0.1.json",
+		                            "@type": "PackageDetails",
+		                            "listed": true,
+		                            "version": "4.0.1"
+		                        }
+		                    }
+		                ]
+		            }
+		        ]
+		    }
+		    `,
+				resultPackageRegistrationPages: []resultPackagePage{},
+				resultPackageSpec: `
+		    <package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
+		<metadata minClientVersion="2.12">
+		      <id>Foo</id>
+		      <version>4.0.1</version>
+		      <title>Foo.NET</title>
+		  </metadata>
+		      </package>
+		    `,
+				version: "4.0.1",
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			p := NewMockpackageManagerClient(ctrl)
+			p.EXPECT().GetURI(gomock.Any()).
+				DoAndReturn(func(url string) (*http.Response, error) {
+					return nugetIndexOrPageTestResults(url, &tt)
+				}).AnyTimes()
+			expectedPackageName := tt.args.expectedPackageName
+			if len(strings.TrimSpace(expectedPackageName)) == 0 {
+				expectedPackageName = tt.args.inputPackageName
+			}
+
+			p.EXPECT().Get(gomock.Any(), expectedPackageName).
+				DoAndReturn(func(url, inputPackageName string) (*http.Response, error) {
+					return nugetPackageIndexAndSpecResponse(t, url, &tt)
+				}).AnyTimes()
+			got, err := fetchGitRepositoryFromNuget(tt.args.inputPackageName, p)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("fetchGitRepositoryFromNuget() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("fetchGitRepositoryFromNuget() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func nugetIndexOrPageTestResults(url string, test *nugetTest) (*http.Response, error) {
+	if url == "https://api.nuget.org/v3/index.json" {
+		if test.wantErr && (test.args.resultIndex == "") {
+			//nolint
+			return nil, errors.New("error")
+		}
+
+		return &http.Response{
+			StatusCode: 200,
+			Body:       io.NopCloser(bytes.NewBufferString(test.args.resultIndex)),
+		}, nil
+	} else {
```

Comment:
probably doesn't need to be an else block if everything in the `if` statement causes a `return`

Reply:
fixed